### PR TITLE
Default the plotting unit to None

### LINKED ISSUE #247: Default the unit of the plotting function to None
As discussed in https://github.com/alchemistry/alchemlyb/issues/244#issuecomment-1264356239
Currently plot_dF_state, plot_ti_dhdl and plot_convergence all have `kT` as default unit. 
This kind of creates extra confusions, as the input estimator or dataframe has already had unit and the plotting function will force them to be `kT`.

I purpose that 

- The unit is default to None and the plotting function will just use the unit already in the estimator or dataframe. If the unit is not None, we would then do the unit conversion.
- Given that we are inputing dataframe to plot_convergence, I'm also thinking of deprecating the unitless array input to plot_convergence to make life easier.

This is partially linked to #246 as currently, the user could change the `delta_f_` and `d_delta_f_`, and then different part of the estimator could have different units. But if the #246 is fixed, we could be sure that the estimator has one unit.

What do you think @orbeckst

### PR DESCRIPTION
**Pull request by xiki-tempula**
Fix #247
Default the unit in `plot_dF_state` , `plot_convergence` and `plot_ti_dhdl` to None.
Remove the array input in `plot_convergence`.

I have fixed the issue here @DrDomenicoMarson

### CODE REVIEW

File: src/alchemlyb/visualisation/convergence.py [hunk 1]
```diff
@@ -5,37 +5,35 @@
...
 
     The input could be the result from
-    :func:`~alchemlyb.convergence.forward_backward_convergence` or it could
-    be given explicitly as `forward`, `forward_error`, `backward`,
-    `backward_error`.
-
-    `forward`: A list of free energy estimate from the first X% of data,
-    where `forward_error` is the corresponding error.
+    :func:`~alchemlyb.convergence.forward_backward_convergence` or
+    :func:`~alchemlyb.convergence.fwdrev_cumavg_Rc`. The input should be a
```
Comment (orbeckst) [inline src/alchemlyb/visualisation/convergence.py:13]: (from other PR on Rc but that's fine)

**Review by orbeckst — CHANGES_REQUESTED** [review]
Looking good but please add entry in CHANGES under Changes

**Comment by xiki-tempula** [comment]
@orbeckst Thanks for the review.
I have added the change log

> Is there a test that shows that (1) None works and that (2) changing units works?

I have added this as well.

**Review by orbeckst — APPROVED** [review]

**Comment by orbeckst** [comment]
thanks!

**Pull request merged**